### DISCUSSION TITLE
Help With Setting Up a Next.js/Java Monorepo with Vercel Deployment

### DISCUSSION THREAD

**Discussion by lewxdev** [vercel/community#1139] [Help] [answered]
### Summary

I'm curious if there's a good way to set up a Next.js frontend with a Java/Spring backend.
Recently inherited/started work on a project using React and Spring, and I love the value in deploying to Vercel, but we'd like to consolidate the two repositories into one. Is there any way to do this and still deploy to Vercel?

**Comment by amyegan** [comment, marked as answer]
Thanks for reaching out @lewxdev! You should have no trouble deploying only the frontend Vercel from a monorepo. I would recommend starting with one of the basic guide in the Vercel docs for [monorepos](https://vercel.com/docs/concepts/monorepos#) if you're trying to figure out how to deploy from a new monorepo. 

If you need help setting up a new monorepo, Turborepo has a nice quickstart guide at [turbo.build/repo/docs](https://turbo.build/repo/docs). They also have a [Discord](https://turbo.build/discord) and [GitHub Discussions](https://github.com/vercel/turbo/discussions) page where you can get more guidance specific to your repo.

Nx has a similar guide available at [nx.dev/getting-started/intro](https://nx.dev/getting-started/intro). They also have [Slack](https://go.nrwl.io/join-slack?utm_source=nx.dev) and [GitHub Discussions](https://github.com/nrwl/nx/discussions) available if you need more specific help.
